From 51d7068f4dad1760b121af678e3ef9f35dae49b0 Mon Sep 17 00:00:00 2001
From: Jeremie Graulle <jeremiex.graulle@intel.com>
Date: Tue, 18 Jun 2013 14:06:33 +0200
Subject: [PORT FROM MAIN] Add Android.mk command to import headers without import symbols

BZ: 117000

LOCAL_COPY_HEADERS is deprecated by Android, we have to use
LOCAL_EXPORT_C_INCLUDE_DIRS instead.
In test we need to import a header without link with the library.
We replace the body of the function by a stub or a mock.
But we need to have the same prototype of the function by importing the
header. But there are no way to import a header exported by a
LOCAL_EXPORT_C_INCLUDE_DIRS without link with the library.

This patch add 2 new LOCAL variables
LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES and
LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES to import include
from libraries without import symbols.

Change-Id: I6f668c900003e27ba3c2ce0d2ab9200b6f058faf
Signed-off-by: Jeremie Graulle <jeremiex.graulle@intel.com>
Reviewed-on: http://android.intel.com:8080/114072
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Rao, Chetan <chetan.rao@intel.com>
Reviewed-by: De Chivre, Renaud <renaud.de.chivre@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 core/binary.mk     |    6 +++++-
 core/clear_vars.mk |    2 ++
 2 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/core/binary.mk b/core/binary.mk
index 92ec69f..f7771d8 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -608,7 +608,11 @@ import_includes_deps := $(strip \
     $(foreach l, $(installed_shared_library_module_names), \
       $(call intermediates-dir-for,SHARED_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE))/export_includes) \
     $(foreach l, $(LOCAL_STATIC_LIBRARIES) $(LOCAL_WHOLE_STATIC_LIBRARIES), \
-      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE))/export_includes))
+      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE))/export_includes) \
+    $(foreach l, $(LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES), \
+      $(call intermediates-dir-for,STATIC_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE))/export_includes) \
+    $(foreach l, $(LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES), \
+      $(call intermediates-dir-for,SHARED_LIBRARIES,$(l),$(LOCAL_IS_HOST_MODULE))/export_includes))
 $(import_includes) : $(import_includes_deps)
 	@echo Import includes file: $@
 	$(hide) mkdir -p $(dir $@) && rm -f $@
diff --git a/core/clear_vars.mk b/core/clear_vars.mk
index 8d84814..49c410e 100644
--- a/core/clear_vars.mk
+++ b/core/clear_vars.mk
@@ -46,6 +46,8 @@ LOCAL_CONLYFLAGS:=
 LOCAL_RTTI_FLAG:=
 LOCAL_C_INCLUDES:=
 LOCAL_EXPORT_C_INCLUDE_DIRS:=
+LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_STATIC_LIBRARIES :=
+LOCAL_IMPORT_C_INCLUDE_DIRS_FROM_SHARED_LIBRARIES :=
 LOCAL_LDFLAGS:=
 LOCAL_LDLIBS:=
 LOCAL_AAPT_FLAGS:=
-- 
1.7.4.1

